tools/jdbc/src/main/java/org/duckdb/DuckDBPreparedStatement.java

```diff
@@ -103,7 +103,12 @@ private void prepare(String sql) throws SQLException {
 			params = new Object[0];
 			// TODO add query type to meta
 			String query_type = DuckDBNative.duckdb_jdbc_prepare_type(stmt_ref);
-			is_update = !query_type.equals("SELECT") && !query_type.equals("PRAGMA") && !query_type.equals("EXPLAIN");
```

Comment:
In C++ the query result has a [statement return type](https://github.com/duckdb/duckdb/blob/master/src/include/duckdb/common/enums/statement_type.hpp#L51) that could be used to determine if the query returns a result. Perhaps a good idea to also use here?

Reply:
One thing to note is that this does not signify the query is an update or not, but only what the query returns. For example, insert statements can also return result sets:

```sql
D create table integers(i integer);
D insert into integers values (42) returning *;
┌────┐
│ i  │
├────┤
│ 42 │
└────┘
```

Reply:
@Mytherin good point, I'll switch it over to that